internal/pkg/util/hasher/hasher.go

```diff
@@ -0,0 +1,44 @@
+package hasher
+
+import (
+	"encoding/binary"
+
+	"golang.org/x/crypto/blake2b"
+)
+
+// Hasher takes in an ordered series of data and hashes it together
+type Hasher struct {
+	toHash [][]byte
+}
+
+// NewHasher creates a new Hasher
+func NewHasher() *Hasher {
+	return &Hasher{
+		toHash: make([][]byte, 0),
+	}
+}
+
+// Int adds a uint64 to the data to be hashed
+func (h *Hasher) Int(i uint64) {
+	intData := make([]byte, binary.MaxVarintLen64)
+	n := binary.PutUvarint(intData, i)
+	h.toHash = append(h.toHash, intData[:n])
+}
+
+// Bytes adds a byte slice to the data to be hashed
+func (h *Hasher) Bytes(bs []byte) {
+	h.toHash = append(h.toHash, bs[:])
+}
+
+// Hash returns the hash value of the provided data
+func (h *Hasher) Hash() []byte {
+	var accum []byte
+	for _, s := range h.toHash {
+		accum = append(accum, s...)
+	}
+	hash := blake2b.Sum256(accum)
```

Comment:
I believe that generally most of the Filecoin hashes, are switching to sha2, including the epost-related ones.